repose-aggregator/components/filters/uri-user-filter/src/test/scala/org/openrepose/filters/uriuser/UriUserFilterTest.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2016 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.filters.uriuser
+
+import javax.servlet.{FilterChain, ServletResponse}
+
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Matchers._
+import org.mockito.Mockito._
+import org.openrepose.commons.utils.http.PowerApiHeader
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.filters.uriuser.config.{IdentificationMapping, IdentificationMappingList, UriUserConfig}
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{Matchers, BeforeAndAfter, FunSpec}
+import org.scalatest.junit.JUnitRunner
+import org.springframework.mock.web.MockHttpServletRequest
+
+@RunWith(classOf[JUnitRunner])
+class UriUserFilterTest extends FunSpec with BeforeAndAfter with Matchers with MockitoSugar {
+
+  import UriUserFilterTest._
+
+  var filter: UriUserFilter = _
+  var request: MockHttpServletRequest = _
+  var response: ServletResponse = _
+  var filterChain: FilterChain = _
+
+  before {
+    request = new MockHttpServletRequest
+    response = mock[ServletResponse]
+    filterChain = mock[FilterChain]
```

Comment:
Would we want to use Spring's MockHttpServletResponse and MockFilterChain here instead? It's more consistent, and you don't have to use an ArgumentCaptor. On the other hand, it may be considered more "heavy-weight" than is necessary.

I don't really mind either way, but thought I would call attention to it nonetheless.